problem_builder/mentoring.py

```diff
@@ -776,3 +780,570 @@ def workbench_scenarios():
         Scenarios displayed by the workbench. Load them from external (private) repository
         """
         return loader.load_scenarios_from_path('templates/xml')
+
+
+@XBlock.needs("i18n")
+@XBlock.wants('settings')
+class MentoringWithExplicitStepsBlock(XBlock, StudioEditableXBlockMixin, StudioContainerXBlockMixin):
```

Comment:
Can we make this a block specifically for handling steps, rather than a copy of the all-encompassing mentoring block? Ie maybe StepManager or StepBuilder?

I haven't looked closely, but it seem there is currently a lot of duplicated code with mentoring, currently? Imho we should only duplicate code we expect to be able to remove from Mentoring once we remove the assessments from there, to handle it exclusively with steps. Can the rest be factorized? Otherwise maintaining twice the "almost the same code" is not going to be fun.